gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -330,10 +382,36 @@ private MapGeoJson toMapGeoJson(GeoJsonFeature feature) {
   }
 
   public LiveData<CameraPosition> getCameraPosition() {
+    checkPointNearVertex(cameraPosition.getValue());
     Timber.d("Current position is %s", cameraPosition.getValue().toString());
     return cameraPosition;
   }
 
+  private void checkPointNearVertex(CameraPosition position) {
+    if (vertices.isEmpty() || vertices.size() < 3) {
+      return;
+    }
+    if (vertices.get(0) == position.getTarget() || isPointNearFirstVertex(position.getTarget())) {
```

Comment:
We should use `equals` instead of actual instance check. I believe the first condition won't be ever true with `=` sign. Also, the second condition includes the first implicitly. So for simplicity, we should just keep the second condition and remove the first one.

Reply:
can replace = with .equals() as that will be checking the content.

Reply:
As mentioned in the previous comment, it's better to remove the equality check entirely and only keep the `isPointNearFirstVertex` for simplicity.

Reply:
removed